### PR TITLE
`@0x:contracts-exchange-libs` Added unit tests to exchange-libs

### PR DESCRIPTION
**Pull request by jalextowle**
## Description

This PR adds unit tests to the `exchange-libs` package. All contracts are covered, and these changes predominately include tests to cover hashing. 

## Testing instructions

`yarn build && yarn test && yarn lint`

## Types of changes

* Unit testing

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: contracts/exchange-libs/contracts/test/TestLibs.sol [hunk 1]
```diff
@@ -173,4 +173,28 @@ contract TestLibs is
         _addFillResults(totalFillResults, singleFillResults);
         return totalFillResults;
     }
+
```
Comment (abandeali1) [inline contracts/exchange-libs/contracts/test/TestLibs.sol]: We should break this up into `TestLibOrder`, `TestLibZeroExTransaction`, etc. #2055 already does this if you want to rebase this PR off of that.

File: contracts/exchange-libs/test/lib_eip712_exchange_domain.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            .generateDomainHash({
+                name: '0x Protocol',
+                version: '3.0.0',
+                chainId: CHAIN_ID,
+                verifyingContractAddress: libsContract.address,
+            })
+            .toString('hex');
+    });
+
+    describe('hashEIP712ExchangeMessage', () => {
```
Comment (abandeali1) [inline contracts/exchange-libs/test/lib_eip712_exchange_domain.ts]: We should specifically test the domain hash too (already done in #2055 :))

File: contracts/exchange-libs/test/lib_zero_ex_transaction.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+                data: randomAssetData(),
+                domain: {
+                    verifyingContractAddress: libsContract.address,
+                    chainId: 1337,
+                },
+            });
+        });
+    });
+
+    describe('hashOrder', () => {
```
Comment (abandeali1) [inline contracts/exchange-libs/test/lib_zero_ex_transaction.ts]: Replace instances of `order` with `zeroExTransaction`

File: contracts/exchange-libs/test/lib_zero_ex_transaction.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+            await testGetTransactionHashAsync({
+                ...EMPTY_TRANSACTION,
+                domain: {
+                    verifyingContractAddress: libsContract.address,
+                    chainId: 1337,
+                },
+            });
+        });
+
+        it('should correctly hash a non-empty order', async () => {
```
Comment (dorothy-zbornak) [inline contracts/exchange-libs/test/lib_zero_ex_transaction.ts]:
```suggestion
        it('should correctly hash a non-empty transaction', async () => {
```

File: contracts/utils/test/lib_eip712.ts [hunk 5]
```diff
@@ -73,6 +54,32 @@ describe('LibEIP712', () => {
         await blockchainLifecycle.revertAsync();
     });
 
+    /**
```
Comment (dorothy-zbornak) [inline contracts/utils/test/lib_eip712.ts]: Can we move this to right before the `_hashEIP712Message` tests and put `testHashEIP712DomainAsync()` in this place?

File: contracts/exchange-libs/test/lib_order.ts [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+        takerFeeAssetData: constants.NULL_BYTES,
+        salt: constants.ZERO_AMOUNT,
+        feeRecipientAddress: constants.NULL_ADDRESS,
+        expirationTimeSeconds: constants.ZERO_AMOUNT,
+    };
+
+    /**
+     * Tests the `_hashOrder()` function against a reference hash.
+     */
+    async function testHashOrderAsync(order: Order): Promise<void> {
```
Comment (dorothy-zbornak) [inline contracts/exchange-libs/test/lib_order.ts]: Would prefer if this was closer to the `hashOrder` tests.

**Pull request merged**